### DISCUSSION TITLE
renaming of 'version' file to something more specific

### DISCUSSION THREAD

**Discussion by atomcraft-th** [nlohmann/json#4243] [Q&A] [answered]
There is a file called 'version' which gets included by macro_scope.hpp.
[Github reference](https://github.com/nlohmann/json/blob/a259ecc51e1951e12f757ce17db958e9881e9c6c/include/nlohmann/detail/macro_scope.hpp#L52)

I have, a couple of times, had issues with this include because it has tried to include my own 'version' file. It doesn't like the format I use in my version file.

If you are going to #include a file for the version, should it perhaps have a less generic name to avoid potential conflict with users code?

Thanks for the consideration.

Tom

**Comment by gregmarr** [comment, marked as answer]
#4241 
If it's including your header instead of the standard library header, then that's because you have the path to your header in the system include path.  You shouldn't do that, or at least don't name your headers the same as standard library headers.

**Comment by atomcraft-th** [comment]
Ah, now I understand, I hadn't realised it was a system header. Thank you.

From what I can tell it only exists from c++20 though. I am using c++17 and so it doesn't exist in the system for me, so if I happen to have that file in my path, it does find it and pick it up.

Thanks for your help. 
If it could be wrapped in a '#if(JSON_HAS_CPP_20)' that would make my life easier, but I can understand how that isn't essential really.

**Reply by gregmarr** [reply]
The `__has_include(<version>)` check should only be looking in the system headers.  What compiler are you using?  Is your `version` file in a directory that is on the compiler include path?
